src/storage/s3/storage.c

```diff
@@ -303,59 +298,20 @@ storageS3Response(HttpRequest *request, StorageS3ResponseParam param)
     ASSERT(request != NULL);
 
     HttpResponse *result = NULL;
-    unsigned int retryRemaining = 2;
-    bool done;
 
-    do
+    MEM_CONTEXT_TEMP_BEGIN()
     {
-        done = true;
-
-        MEM_CONTEXT_TEMP_BEGIN()
-        {
-            // Process request
-            result = httpRequest(request, !param.contentIo);
-
-            // Error if the request was not successful
-            if (!httpResponseCodeOk(result) && (!param.allowMissing || httpResponseCode(result) != HTTP_RESPONSE_CODE_NOT_FOUND))
-            {
-                // If there are retries remaining and a response parse it as XML to extract the S3 error code
-                const Buffer *content = httpResponseContent(result);
+        // Get response
+        result = httpRequest(request, !param.contentIo);
 
-                if (bufUsed(content) > 0 && retryRemaining > 0)
-                {
-                    // Attempt to parse the XML and extract the S3 error code
-                    TRY_BEGIN()
-                    {
-                        XmlNode *error = xmlDocumentRoot(xmlDocumentNewBuf(content));
-                        const String *errorCode = xmlNodeContent(xmlNodeChild(error, S3_XML_TAG_CODE_STR, true));
-
-                        if (strEq(errorCode, S3_ERROR_REQUEST_TIME_TOO_SKEWED_STR))
-                        {
-                            LOG_DEBUG_FMT(
-                                "retry %s: %s", strPtr(errorCode),
-                                strPtr(xmlNodeContent(xmlNodeChild(error, S3_XML_TAG_MESSAGE_STR, true))));
-
-                            retryRemaining--;
-                            done = false;
-                        }
-                    }
-                    // On failure just drop through and report the error as usual
-                    CATCH_ANY()
-                    {
-                    }
-                    TRY_END();
-                }
+        // Error if the request was not successful
+        if (!httpResponseCodeOk(result) && (!param.allowMissing || httpResponseCode(result) != HTTP_RESPONSE_CODE_NOT_FOUND))
+            httpRequestError(request, result);
```

Comment:
If we're in the expire command and we're requesting to delete a file from S3, allowMissing is true and the HTTP response S3_ERROR_REQUEST_TIME_TOO_SKEWED_STR, then we'll no longer be retrying since the retries have been removed from this code and Expire is not part of the cmdLocal - am I reading that right?

Reply:
You are correct, but that's true of a lot of other errors as well. Really, we should not be getting the clock skew error anymore. If we are it's a sign of something really bad happening. Also, the clock skew error was only happening when writing files -- not while deleting/listing, etc.